server/src/test/java/io/crate/types/GeoPointTypeTest.java

```diff
@@ -28,13 +28,22 @@
 
 import org.elasticsearch.common.io.stream.BytesStreamOutput;
 import org.elasticsearch.common.io.stream.StreamInput;
-import org.elasticsearch.test.ESTestCase;
 import org.junit.Test;
 import org.locationtech.spatial4j.context.jts.JtsSpatialContext;
 import org.locationtech.spatial4j.shape.Point;
 import org.locationtech.spatial4j.shape.impl.PointImpl;
 
-public class GeoPointTypeTest extends ESTestCase {
+public class GeoPointTypeTest extends DataTypeTestCase<Point> {
+
+    @Override
+    public DataType<Point> getType() {
+        return GeoPointType.INSTANCE;
+    }
+
+    @Override
+    protected boolean supportsDocValues() {
```

Comment:
This probably needs a better name, as the data type may support doc values, it just doesn't store them in the expected way.  Another alternative might be to provide a function that converts the read values so that they can be properly compared with the input, with a default identity implementation.  However, object types need to do something different here.